File: guides/java/kora-java-guide-json-app/src/test/java/ru/tinkoff/kora/guide/json/JsonAppTest.java
package ru.tinkoff.kora.guide.json;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertInstanceOf;

import org.junit.jupiter.api.Test;
import ru.tinkoff.kora.guide.json.dto.UserRequest;
import ru.tinkoff.kora.guide.json.dto.UserResult;
import ru.tinkoff.kora.guide.json.service.UserService;
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest;
import ru.tinkoff.kora.test.extension.junit5.TestComponent;

@KoraAppTest(Application.class)
class JsonAppTest {

    @TestComponent
    private UserService userService;

    @Test
    void sealedJsonResponseFlow() {
        var created = userService.createUser(new UserRequest("Alice", "alice@example.com"));
        var result = userService.getUser(created.id());

        var success = assertInstanceOf(UserResult.UserSuccess.class, result);
        assertEquals(UserResult.Status.OK, success.status());
        assertEquals("Alice", success.user().name());
    }
}

File: guides/kotlin/kora-kotlin-guide-json-app/src/test/kotlin/ru/tinkoff/kora/guide/json/JsonAppTest.kt
package ru.tinkoff.kora.guide.json

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import ru.tinkoff.kora.guide.json.dto.UserRequest
import ru.tinkoff.kora.guide.json.dto.UserResult
import ru.tinkoff.kora.guide.json.service.UserService
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest
import ru.tinkoff.kora.test.extension.junit5.TestComponent

@KoraAppTest(Application::class)
class JsonAppTest {
    @TestComponent
    lateinit var userService: UserService

    @Test
    fun componentsAreWired() {
        assertNotNull(userService)
    }

    @Test
    fun sealedJsonResponseFlow() {
        val created = userService.createUser(UserRequest("Alice", "alice@example.com"))
        val result = userService.getUser(created.id)

        val success = assertInstanceOf(UserResult.UserSuccess::class.java, result)
        assertEquals(UserResult.Status.OK, success.status)
        assertEquals("Alice", success.user.name)
    }
}
